bundle/src/main/java/com/adobe/acs/commons/users/impl/EnsureAce.java

```diff
@@ -240,21 +249,23 @@ public void removeAces(ResourceResolver resourceResolver, Authorizable authoriza
 
         final Query query =
                 queryBuilder.createQuery(PredicateGroup.create(params), resourceResolver.adaptTo(Session.class));
-        final Iterator<Resource> resources = query.getResult().getResources();
-
-        while (resources.hasNext()) {
-            // Get the content resource as this is what the AccessControlManager API will use to find the ACLs under it
-            Resource contentResource = resources.next().getParent().getParent();
-
-            if (!paths.contains(contentResource.getPath())) {
-                for (AccessControlPolicy policy : accessControlManager.getPolicies(contentResource.getPath())) {
-                    if (policy instanceof JackrabbitAccessControlList) {
-                        acls.add((JackrabbitAccessControlList) policy);
-                        break;
+        query.getResult().getHits().forEach(hit -> {
+            try {
+                Resource aceResource = hit.getResource();
```

Comment:
Oops... so this is going to be problematic.. we want this to me;:

```
query.getResult().getHits().forEach(hit -> {
   try {  
           Resource aceResource = resourceResolver.getResource(hit.getPath());
   ...
```

The problem is whenever we call a "Sling" (getResource or getProperties) method on the QB results, its creates the leaking RR, so we want to make sure we never do that, and instead use OUR RRs which we ensure is closed properly.

Reply:
ah, got it.  that makes sense...will fix it later tonight.